src/tequila/circuit/_gates_impl.py

```diff
@@ -268,14 +282,35 @@ def __pow__(self, power, modulo=None):
 
 
 class PowerGateImpl(ParametrizedGateImpl):
+    """
+    Attributes
+    ---------
+    power
+        numeric type (fixed exponent) or hashable type (parametrized exponent)
+    parameter
+        power multiplied by pi
+        to be consitent with exp(-i a/2 G) representation [a: gate.parameter, G: gate.generator]
+    """
+
+    @property
+    def power(self):
+        return self._power
+
+    @power.setter
+    def power(self, other):
+        self._power = assign_variable(variable=other)
```

Comment:
needs to set self.parameter as well to stay consistent

Reply:
see for example dagger operation

Reply:
@georgios-ts 👆